extension/src/panel.ts

```diff
@@ -31,22 +32,20 @@ async function fileType(path: string): Promise<vscode.FileType | undefined> {
     const stat = await vscode.workspace.fs.stat(vscode.Uri.file(path))
     return stat.type
   } catch (err) {
-    // TODO: check exception and throw if unexpected
-    return undefined
+    throw new Error(`${err}`)
   }
 }
 
 export class HistoryDebuggerPanel {
   protected static _instance?: HistoryDebuggerPanel
 
-  static install(
-    extensionUri: vscode.Uri,
-    secretStorage: vscode.SecretStorage,
-    httpServerUrl: string,
-  ): HistoryDebuggerPanel {
+  static async install(extensionUri: vscode.Uri, secretStorage: vscode.SecretStorage): Promise<HistoryDebuggerPanel> {
     if (this._instance === undefined) {
-      this._instance = new this(extensionUri, secretStorage, httpServerUrl)
+      const server = await Server.create()
+      console.log(`Server listening on ${server.url}`)
+      this._instance = new this(extensionUri, secretStorage, server)
```

Comment:
You're ignoring the (very rare) chance of a race where an instance already being created and server is not yet ready.
In this race, we'll end up with multiple instances of the panel.

You can use this by storing the creation Promise statically on the class and checking its existence instead of checking `_instance`.

Reply:
Thank you for the reminder here! Somehow missed this in my notes from our meeting.